sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -226,6 +258,41 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
     result.copy(expectedOutputAttributes = Some(metastoreRelation.output))
   }
 
+  private def inferSchema(
+      metastoreSchema: StructType,
+      options: Map[String, String],
+      fileFormat: FileFormat,
+      fileType: String,
+      fileIndex: FileIndex): Option[StructType] = {
+    val inferred = fileFormat.inferSchema(
+      sparkSession,
+      options,
+      fileIndex.listFiles(Nil).flatMap(_.files))
+    if (fileType.equals("parquet")) {
+      inferred.map(ParquetFileFormat.mergeMetastoreParquetSchema(metastoreSchema, _))
+    } else {
+      inferred
+    }
+  }
+
+  private def updateCatalogTable(
+      catalogTable: CatalogTable,
+      inferredSchema: Option[StructType]): Option[CatalogTable] = try {
+    inferredSchema.flatMap { schema =>
```

Comment:
We want the return value of this operation. It'll be the updated ```CatalogTable``` record if the update is successful.